server/src/test/java/org/elasticsearch/search/SearchHitsTests.java

```diff
@@ -21,88 +21,126 @@
+        for (int i = 0; i < hits.length; i++) {
+            hits[i] = SearchHitTests.createTestItem(xContentType, withOptionalInnerHits, withShardTarget);
         }
+        return hits;
+    }
+
+    private static TotalHits randomTotalHits() {
         long totalHits = TestUtil.nextLong(random(), 0, Long.MAX_VALUE);
         TotalHits.Relation relation = randomFrom(TotalHits.Relation.values());
+        return new TotalHits(totalHits, relation);
+    }
+
+    public static SearchHits createTestItem(XContentType xContentType, boolean withOptionalInnerHits, boolean withShardTarget) {
+        int searchHits = randomIntBetween(0, 5);
+        SearchHit[] hits = createSearchHitArray(searchHits, xContentType, withOptionalInnerHits, withShardTarget);
         float maxScore = frequently() ? randomFloat() : Float.NaN;
+        return new SearchHits(hits, frequently() ? randomTotalHits() : null, maxScore);
+    }
 
-        return new SearchHits(hits, frequently() ? new TotalHits(totalHits, relation) : null, maxScore);
+    @Override
+    protected SearchHits mutateInstance(SearchHits instance) {
+        switch (randomIntBetween(0, 2)) {
+            case 0:
+                return new SearchHits(createSearchHitArray(instance.getHits().length + 1,
+                    randomFrom(XContentType.values()), false, randomBoolean()),
+                    instance.getTotalHits(), instance.getMaxScore());
+            case 1:
+                final TotalHits totalHits;
+                if (instance.getTotalHits() == null) {
+                    totalHits = randomTotalHits();
+                } else {
+                    totalHits = null;
+                }
+                return new SearchHits(instance.getHits(), totalHits, instance.getMaxScore());
+            case 2:
+                final float maxScore;
+                if (Float.isNaN(instance.getMaxScore())) {
+                    maxScore = randomFloat();
+                } else {
+                    maxScore = Float.NaN;
+                }
+                return new SearchHits(instance.getHits(), instance.getTotalHits(), maxScore);
+            default:
+                throw new UnsupportedOperationException();
+        }
+    }
+
+    @Override
+    protected Predicate<String> getRandomFieldsExcludeFilter() {
+        return path -> (path.isEmpty() ||
+            path.contains("inner_hits") || path.contains("highlight") || path.contains("fields") || path.contains("_source"));
+    }
+
+    @Override
+    protected String[] getShuffleFieldsExceptions() {
+        return new String[] {"_source"};
     }
 
     @Override
     protected SearchHits createBlankInstance() {
-        return new SearchHits(new SearchHit[0], new TotalHits(0, TotalHits.Relation.EQUAL_TO), Float.NaN);
+        return new SearchHits();
     }
 
     @Override
     protected SearchHits createTestInstance() {
-        return createTestItem();
+        return createTestItem(randomFrom(XContentType.values()), true, true);
```

Comment:
would you mind adding a comment here and in the other createXContentInstance method to explain why the two boolean flags are set that way?

Reply:
Sure, done in https://github.com/elastic/elasticsearch/pull/36313/commits/d6e724a6520bab25597bb6859c139dda674f85e3